### ISSUE TITLE
Regression - Globalize not anymore thread safe

### ISSUE THREAD

**Issue by mtparet** [globalize/globalize#390]
Because of https://github.com/globalize/globalize/pull/374, now we can't anymore use Globalize in threaded contexts like Sidekiq. (Because now the current locale is shared across all threads instead of being specific to the current thread)

Seems like an important breaking change.

Following test succeeded before the commit removing usage of `Thread.current` :
```ruby
  describe 'with locale' do
    def perform_with_locale(locale)
      Thread.new do
        Globalize.with_locale(locale) do
          sleep 1
          puts "#{Thread.current} : locale #{locale}, fallback: #{Globalize.locale} \n"
          expect(Globalize.locale).to eq(locale)
        end
      end
    end

    it 'succeed in one thread' do
      thread = perform_with_locale(:end)
      thread.join
    end

    it 'succeed multithreading' do
      thread1 = perform_with_locale(:en)

      thread2 = perform_with_locale(:fr)

      thread3 = perform_with_locale(:de)

      thread1.join && thread2.join && thread3.join
    end
  end
```

**Comment by Dschee** [comment]
Would love to see this fixed, it is a really annoying bug in combination with sidekiq! I find #374 to be less important than this and therefore hope that @hube's fix is reverted until someone finds a better solution. 

Globalize is pretty useless for me if I can't update any data in more than one language in the background with sidekiq. @parndt Could you have a look and revert if possible?

**Comment by simi** [comment]
@Dschee @hube :( I haven't problem with `Thread.current`. But maybe some config value would be good here since someone will need to use https://github.com/steveklabnik/request_store for example. Any idea for API?

**Comment by scarfacedeb** [comment]
Fixed in #420
